airflow/ui/src/pages/DagsList/DagsFilters.tsx

```diff
@@ -51,6 +57,21 @@ export const DagsFilters = () => {
       [pagination, searchParams, setSearchParams, setTableURLState, sorting],
     );
 
+  const handleStateChange: React.MouseEventHandler<HTMLButtonElement> =
+    useCallback(
+      ({ target }) => {
+        const { value } = target as HTMLButtonElement;
```

Comment:
```suggestion
  const handleStateChange: React.MouseEventHandler<HTMLButtonElement> =
    useCallback(
      ({ currentTarget: { value } }) => {
```

This should remove the need to use `as`